src/sentry/api/bases/organization.py

```diff
@@ -159,6 +163,38 @@ class OrganizationAlertRulePermission(OrganizationPermission):
 class OrganizationEndpoint(Endpoint):
     permission_classes = (OrganizationPermission,)
 
+    def initialize_request(self, request: Request, *args, **kwargs):
+        if request.user and request.user.is_superuser and is_active_superuser(request):
+            organization_slug = kwargs.get("organization_slug")
+            if not request.session.get("su_orgs_accessed"):
+                request.session["su_orgs_accessed"] = [organization_slug]
```

Comment:
Don't we skip logging in this case?

Reply:
The purpose of this feature is to track when our superuser's access different orgs when they have an active superuser session, This logging is only for when superuser's access other orgs. 

I don't believe that user's accessing functionality that they don't have permissions for are within the scope of the project.

As for the middleware, i thought of the middleware approach too, but decided that this would be good for the purposes if this project. i also heard Aniket and Volo had trouble with accessing orgs through the middleware so im not too sure

Reply:
Where do we check that the user is accessing an org that they normally wouldn't have access to? It looks like we'll log whenever someone has SU and access an org?

Reply:
actually, will look into the middleware approach for this

Reply:
> Where do we check that the user is accessing an org that they normally wouldn't have access to? It looks like we'll log whenever someone has SU and access an org?

good point, right now when the user has SU we log it every time they visit an unique org, should we only log it if they access orgs they normally wouldn't have access to or all unique orgs they visit in the SU session?

Reply:
Middleware might be hard since I don't know if `request.access` has been set yet.

I don't really know the requirements of this project, but to me it seems like we want to log when users access orgs that they wouldn't usually have access to via SU. You'd have to look into how permissions work in detail, but looking at https://github.com/getsentry/sentry/blob/bf0007acaaecdb5bfc0b2a93dc8cd5be6b70a80c/src/sentry/auth/access.py#L252-L273, I think that if you check `request.access.role is None` then you can tell that the user doesn't have a role in that org (please double check this).

I think that this logic might also work well in `dispatch`, https://github.com/getsentry/sentry/blob/214ab88c5fb4e4e307138d5e91798b36e11153c3/src/sentry/api/base.py#L194, so that it works for all endpoints vs just `Organization` specific ones.

Reply:
Actually... maybe the logic for logging this just belongs in `from_request` in `access.py`

Reply:
after discussion, we decided to go with logging whenever someone has SU and access an org over only when they access an org they normally don't have access to. this was due to when granted super user, the user will get permissions for things they normally don't have permissions to do so in the cases of them gaining additional permissions on an org they have access to, we want to still log that

Reply:
> > lgtm. What's the intent of this logging right now? Is this intended as an audit trail?
> 
> yup, at the end, we want to have a dashboard where we can track who/why requested superuser and what orgs they accessed during that superuser session.

Ok, putting this in the logger might be kind of a slow way for this dashboard to work, but it can be a good first step. Filtering to logs in gcp is reasonably slow.